ops/jenkins/bfd-jenkins-sonarqube/Jenkinsfile

```
@@ -69,21 +72,30 @@ spec:
             // Build Step
          steps {
                 script {
+                    // Address limitations resulting from CVE-2022-2476
+                    sh "git config --global --add safe.directory $WORKSPACE"
                    // Load the child Jenkinsfiles.
                    scriptForApps = load('apps/build.groovy')
                    //Authentication of AWS to run Build.groovy
                    awsAuth.assumeRole()
                    // Perform Build steps
                    scriptForApps.build(verboseMaven, true)
+
+                   //Retrieve Sonar Token from AWS
+                    def parameterValue = sh(
+                             script: """ aws ssm get-parameter --name "${ACCESS_KEY_PATH}" --region us-east-1 --with-decryption --query 'Parameter.Value' --output text""",
+                             returnStdout: true
+                        ).trim() as String
+                    // Set the token as an environment variable
+                    SONAR_TOKEN = parameterValue
```

Hmm. Probably something like this?
```suggestion
                    env.SONAR_TOKEN = awsSsm.getParameter(parameterName: "/bfd/mgmt/common/sensitive/service_accounts/sonar/service_account_access_key")
```

---

Yh I can do that. I'm going to have to edit the `the getParameter` method in `awsSsm.groovy` to pass in a `isSensitive` feature flag. This will allow us decrypt the value using the `--with-decryption` flag. Another approach could to be to just update the method with `--with-decryption` since it wouldnt matter for nonsensitive parameters. thoughts @mjburling 

- awsSSm.groovy Update.
```
String getParameter(Map args = [:]) {
    name = args.parameterName
    awsRegion = args.awsRegion ?: 'us-east-1'
    isSensitive = args.isSensitive ?: false

    if(isSensitive) {
        return sh(returnStdout: true, script: "aws ssm get-parameter --name ${name} --region ${awsRegion} --with-decryption --query 'Parameter.Value' --output text").trim()
    }
    return sh(returnStdout: true, script: "aws ssm get-parameter --name ${name} --region ${awsRegion} --query 'Parameter.Value' --output text").trim()
}
                  
```